[16.0][MIG] l10n_br_base

mileo opened the pull request:
Testing https://github.com/OCA/oca-port

OCA-git-bot committed:
[ADD] setup.py

antoniospneto committed:
[FIX] l10n_br_base: bad address in demo partner

antoniospneto committed:
[REF] remove unnecessary code involving the address

oca-travis committed:
[UPD] Update l10n_br_base.pot

OCA-git-bot committed:
l10n_br_base 14.0.2.3.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_base
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_base/

douglascstd committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 99.3% (5368 of 5401 strings)

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_base
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_base/pt_BR/

antoniospneto committed:
[IMP] l10n_br_base: add pix fields for partners

rvalyi committed:
[DOC] README + CONTRIBUTORS update

Commit:
[UPD] Update l10n_br_base.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
l10n_br_base 14.0.3.0.0

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_base
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_base/

antoniospneto committed:
[FIX] l10n_br_base: missing licence

OCA-git-bot committed:
l10n_br_base 14.0.3.0.1

rvalyi reviewed (dismissed):
@mileo prós 8-10 módulos mais básicos do repo, (como esse) seria ideal fazer que nem na v13, fazer uma etapa intermediária na v15. Quase que não custa nada de fazer a migração em 2 passos mas manda um sinal bem melhor para quem se ferrou com a v15 (por exemplo através de uma implementação feita por um parceiro oficial; não é pouca gente que tá se ferrando) e que assim poderia enxergar alguma saída pela OCA (já que não poderia migrar para a v16 já). Do nosso lado a gente não pretende manter nada na v15, mas TB não iremos blocar caso alguém cuida de fazer os ports e backports na v15. E mesmo se a v15 tiver apenas 10 ou algum dia talvez até 20 módulos já seria alguma para essa galera que foi se perder na v15 sem procurar sobre o projeto antes.

Alem disso seria bom segurar um pouquinho a onda da 16.0 para limpar ainda alguns detalhes na 14.0 (inclusive que vem de vcs). Pois a partir do momento que a gente começa a fazer os merges na 16.0 para valer, considerando os nossos recursos ficaria inviavel fazer backports e frontports com a 12.0 mais. Manter 2 versões ja é o máximo que a gente consegue fazer e não seria bem vc que poderia demostrar o contrario (estão mais de um ano sem revisar quase nada nem resolver nenhum bug no projeto). Então algumas semanas ou meses ainda antes de desligar a 14.0 da 12.0 não faria mal. Se vc ainda considerar que vai levar alguns meses para ter o modulo account_reconciliation_widget na 16.0 tb não adiantaria muito ter os módulos de localização do account tão cedo (o base poderia ate ser um pouco antes). Quem tiver com fogo no rabo pode ajudar na migração do account_reconciliation_widget eu acho que seria muito bem vindo.

cc @renatonlima @marcelsavegnago @netosjb @felipemotter @mbcosta

mileo committed:
[REM] String definition

mileo committed:
[MIG] base_address_city merged with base_address_extended

mileo committed:
[REM] l10n_br_base transalation hook

mileo committed:
[MIG] To new assets format

mileo committed:
[ADD] Street number

mileo committed:
[REF] Address format

mileo committed:
[REF] Partner Fields size

mileo commented:
> @mileo prós 8-10 módulos mais básicos do repo, (como esse) seria ideal fazer que nem na v13, fazer uma etapa intermediária na v15. Quase que não custa nada de fazer a migração em 2 passos mas manda um sinal bem melhor para quem se ferrou com a v15 (por examplo através de uma implementação feita por um parceiro oficial; não é pouca gente que tá se ferrando) e que assim poderia enxergar alguma saída pela OCA (já que não poderia migrar para a v16 já). Do nosso lado a gente não pretende manter nada na v15, mas TB não iremos blocar caso alguém cuida de fazer os ports e backports na v15. E mesmo se a v15 tiver apenas 10 ou algum dia talvez até 20 módulos já seria alguma para essa galera que foi se perder na v15 sem procurar sobre o projeto antes.
> 
> Alem disso seria bom segurar um pouquinho a onda da 16.0 para limpar ainda alguns detalhes na 14.0 (inclusive que vem de vcs). Pois a partir do momento que a gente começa a fazer os merges na 16.0 para valer, considerando os nossos recursos ficaria inviavel fazer backports e frontports com a 12.0 mais. Manter 2 versões ja é o máximo que a gente consegue fazer e não seria bem vc que poderia demostrar o contrario (estão mais de um ano sem revisar quase nada nem resolver nenhum bug no projeto). Então algumas semanas ou meses ainda antes de desligar a 14.0 da 12.0 não faria mal. Se vc ainda considerar que vai levar alguns meses para ter o modulo account_reconciliation_widget na 16.0 tb não adiantaria muito ter os módulos de localização do account tão cedo (o base poderia ate ser um pouco antes). Quem tiver com fogo no rabo pode ajudar na migração do account_reconciliation_widget eu acho que seria muito bem vindo.
> 
> cc @renatonlima @marcelsavegnago @netosjb @felipemotter @mbcosta

@rvalyi não iremos nesse momento investir em uma versão intermediária, como a 15.0, a não ser que tenha algum cliente que invista nisso. 

Nosso foco nesses próximos dias será revisar os PRs da v14 e migrar os módulos básicos da localização para a v16.

mileo committed:
[REF] do not import tests l10n_br_base

mileo committed:
[REF] str-format-used l10n_br_base

Commit:
[UPD] Update l10n_br_base.pot

OCA-git-bot committed:
l10n_br_base 14.0.3.0.2

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-brazil-14.0/l10n-brazil-14.0-l10n_br_base
Translate-URL: https://translation.odoo-community.org/projects/l10n-brazil-14-0/l10n-brazil-14-0-l10n_br_base/

mileo committed:
[REF] raise-missing-from l10n_br_base

OCA-git-bot committed:
l10n_br_base 14.0.3.0.3

mileo commented:
/ocabot migration l10n_br_base

rvalyi commented on l10n_br_base/models/res_partner.py:
```diff
@@ -20,6 +20,16 @@ class Partner(models.Model):
 
+    def _inverse_street_data(self):
+        """update self.street based on street_name, street_number and street_number2"""
+        for partner in self:
```
A gente ta se esforçando para que os módulos centrais da localização funcionam com bancos multi-pais, nisso
tem que filtrar apenas os registros pros quais partner.country_id.code.upper() == "BR" e senão chamar super.

Eu tb acho melhor falar no docstring o que a gente faz no override do que fazer o papagaio do docstring original.
Poderia ser algo do tipo:
""'"In Brazil the address format is street_name, street_number (comma instead of space)"""

mileo replied:
Solved

mileo committed:
[IMP] Partner street for Brazilian partners and others

mileo committed:
[ADD] street_name to res.company

douglascstd reviewed (changes requested):
[REPROVADO] Testes Funcionais
Não aparece na view do contact o campo para Instant Payment Method (PIX)
![image](https://user-images.githubusercontent.com/20867090/197548994-9b6c0215-23ac-44ca-853d-5f27b8092214.png)

### APROVADO
#### Outros testes realizados:
Inserção e validação do CNPJ:
![image](https://user-images.githubusercontent.com/20867090/197549175-27c4d72a-1746-4edd-85df-182d3af9d676.png)

#### Configuração Localicalização Brasil:
![image](https://user-images.githubusercontent.com/20867090/197549265-cab96c07-651a-4135-a3e6-648ac3047aa1.png)

#### Inserção e validação da inscrição estadual
![image](https://user-images.githubusercontent.com/20867090/197549488-bcd028e0-d246-4ae0-99df-d3a12d38f356.png)

mileo commented:
> [REPROVADO] Testes Funcionais Não aparece na view do contact o campo para Instant Payment Method (PIX) ![image](https://user-images.githubusercontent.com/20867090/197548994-9b6c0215-23ac-44ca-853d-5f27b8092214.png)
> 
> ### APROVADO
> #### Outros testes realizados:
> Inserção e validação do CNPJ: ![image](https://user-images.githubusercontent.com/20867090/197549175-27c4d72a-1746-4edd-85df-182d3af9d676.png)
> 
> #### Configuração Localicalização Brasil:
> ![image](https://user-images.githubusercontent.com/20867090/197549265-cab96c07-651a-4135-a3e6-648ac3047aa1.png)
> 
> #### Inserção e validação da inscrição estadual
> ![image](https://user-images.githubusercontent.com/20867090/197549488-bcd028e0-d246-4ae0-99df-d3a12d38f356.png)

@douglascstd  obrigado pela revisão, mas esse campo é adicionado somente pelo módulo l10n_br_account:

https://github.com/OCA/l10n-brazil/blob/14.0/l10n_br_account/views/res_partner_view.xml#L21-L28

rvalyi commented:
@mileo parece OK ou quase, mas como eu comentei eu acharia melhor fazer um PR intermediário na 15.0 para esse e mais uns 3 ou 4 módulos. O trabalho da migração sendo feito, é muito fácil dividir os commits que iriam num PR de migração de 15 e num outro de 16. Os argumentos para isso são:

- O "big picture" é que a Odoo SA e uma turma da "localização do ano" tentam descreditar a OCA. So que ao mesmo tempo 90% ou mais das empresas que tem scope grandes e que estão tentando implementar com as localizações piratas estão quebrando a cara. No que a gente consegue extrapolar deve ser dezenas e dezenas que investiram dezenas de milhares de R$ para ficar sem saída na 14 ou na 15 (os caras entram em contato de forma privada). Uma grande parte deles não tem condição de migrar para a v16 agora (por conta das customizações, por conta da falta de eco sistema, da falta de grana, de não existir scripts OpenUpgrade para a 16 tão cedo etc...). Nisso a gente pensa que é mandar um sinal muito melhor de ter alguns módulos OCA na 15. O TI que fuça vai ver que o o modulo fiscal da OCA existe para todas versões do Odoo, ai isso é mostrar um diferencial forte com a localização X do parceiro gold de 2021 que é diferente da localização do parceiro gold de 2022 que copiou do X mas re-empacotou enquanto o X ta tomando caldo.

- Tecnicamente falando, migrar a maioria dos módulos da localização da 14 para a 16 é trivial. 10 vezes menos trabalho do que a migração v12 v14. Porem no caso do l10n_br_account e planos de conta talvez não seja tão trivial assim. Ai quebrar em 2 passos com os testes passando na 15 primeiro fica realmente melhor. Pela gente, não precisa passar do l10n_br_account ou l10n_br_nfe na 15. Tem caso que o cara precisa so calcular e lançar os impostos ou so emitir NFe para casos triviais. Ai se ele quiser mais ele contrata alguém para migrar módulos ou migrar para a v16, mas pelo menos a gente estende a mão e não parece se igualar a falta de continuidade que tem fora da OCA.

Por fim, a gente consegue ajudar nisso. Eu não tenho tempo agora. Mas de noite eu consigo ajeitar os commits para fazer esse PR intermediário na 15.0 e o PR da 16 que sai da 15.0.

douglascstd reviewed (approved):
Entendido sobre a view do PIX que não está aparente neste módulo base.

mileo committed:
[MIG][l10n_br_base] bump module version 15.0.1.0.0

mileo committed:
[REF] l10n_br_base res.config view

rvalyi commented:
@douglascstd @gabrielcardoso21 se puderem testar essa na 15 https://github.com/OCA/l10n-brazil/pull/2194 que ai a gente passa pela 15 primeiro fica melhor para esses módulos super centrais.

Commit:
[UPD] Update l10n_br_base.pot

OCA-git-bot committed:
[UPD] README.rst

mileo committed:
[IMP] l10n_br_base: black, isort, prettier

mileo committed:
[MIG][l10n_br_base] bump module version

mileo commented:
/ocabot merge nobump

Histórico reescrito com PR da v15

rvalyi commented:
IMPORTANTE: o procedimento tecnico para fazer o PR não foi correto e isso perdeu mais de 550 commits. Explicamos e corrigimos o problema [aqui](#3490)
